### ISSUE TITLE
Clipper w/images: y-axis not flipped, and transform coordinates are flipped

### ISSUE THREAD

**Issue by EricThomson** [vispy/vispy#1564]
### tl;dr
1.  `Clipper()`, which I use to make a clipping mask for multiple images, uses an unflipped y-axis. This is bad for `ImageVisual`s and yields some strange behavior (e.g., images being cut off when you resize).
2. When applying transforms to the clipping window, x/y seem flipped (e.g.,+x translation moves it to the left).

Is there some way I'm missing to change these? I've gone into the source code and didn't find anything obvious (I discuss `Rect.flipped()` below).

### Too long never read
I am using the clipping mask on images. Here is a minimal example:

    import numpy as np
    import vispy.app
    from vispy import gloo, visuals
    from vispy.visuals.filters import Clipper
    from vispy.visuals.transforms import MatrixTransform()

    # Create image
    Z = np.zeros((300,300), dtype = np.uint8)
    Z[200:205,:] = 200 #light line
    clip_rect = [50, 10, 200, 100]  #x,y,w,h

    # Display
    class Canvas(vispy.app.Canvas):
        def __init__(self):
            vispy.app.Canvas.__init__(self, keys='interactive', size= (300, 300))
            self.image = visuals.ImageVisual(Z, cmap = 'grays', method='subdivide')
            clip = Clipper(clip_rect)

            #insert transforms here

            self.image.attach(clip)
            self.show()

        def on_draw(self, ev):
            gloo.clear(color='gray', depth=True)
            self.image.draw()

        def on_resize(self, event):
            vp = (0, 0, self.physical_size[0], self.physical_size[1])
            self.context.set_viewport(*vp)
            self.image.transforms.configure(canvas=self, viewport=vp)

    if __name__ == '__main__':
        win = Canvas()
        import sys
        if sys.flags.interactive != 1:
            vispy.app.run()

I noticed a few things when playing with this simple example:

#### Clipper coordinates are at bottom left
One, the clipper coordinates start at the bottom-left. This is unnatural for images, and I wonder if this could be changed. There was a similar issue that was solved with `view.camera.flip = (0, 1, 0)`, but I can't find similar ways to flip the y axis for this use case (https://github.com/vispy/vispy/issues/904).

There is a way to flip the [`Rect`](https://github.com/vispy/vispy/blob/master/vispy/geometry/rect.py) that is created within [`clipper.py`](https://github.com/vispy/vispy/blob/master/vispy/visuals/filters/clipper.py), but this  doesn't flip the coordinates it just flips values *within* the unflipped coordinate system.

One reason this choice of origin is a problem is that when you resize the canvas, the clipping window *slides* around on the image, so you see different parts of the image, and when you expand downward the image eventually completely disappears. This is unusual default behavior, you can easily reproduce with my code above.

#### Clipper translation transforms seem flipped
Clippers can take in transforms, which is awesome. However, it seems the transforms are flipped in x/y: if you translate in +x, it shifts the clipping mask to the left. If you move in +y, it moves things down (which is what I want, but flipped relative to the convention used by the clipper). If you rotate, it rotates relative to the bottom left origin, again something that is unnatural when working with images, but I just thought I'd note it.

You can test out transforms by putting the following into the above code where it says 'insert transforms here':

    tr = MatrixTransform()
    #tr.rotate(45, (0,0,1))
    tr.translate((45, -50)) #x_translate, y_translate
    clip.transform = tr

#### What I plan to do
I can work with the constraints set out above, repeating calculations for my clipping mask and for my images. However, I am hoping I am missing something obvious, some toggle that will just make all this work, because I *really* love DRY in my Python. :)
